### DISCUSSION TITLE
NASA's Astronomy Image Of The Day

### DISCUSSION THREAD

**Discussion by Pashtunzoy** [johnlindquist/kit#677] [Share]
A simple script that shows you info about NASA's astronomy image of the day.

[Open in ScriptKit](https://scriptkit.com/api/new?name=nasa-astronomy-pic-of-the-day&url=https://gist.githubusercontent.com/Pashtunzoy/e6cc24632eb237d3001df96c8253032d/raw/adc04ee8ccdaf4f726c1b300480da5a5bd03486e/nasa-astronomy-pic-of-the-day.js)

Example:
![nasa-astronomy-pic-of-the-day-500](https://user-images.githubusercontent.com/12707591/147915697-5744c6bf-e9a2-4d75-84de-ac88855754bc.png)

## The Code:
```js
// Menu: NASA's Astronomy Picture of the Day
// Description: Get NASA's APOTD
// Suggested Schedule cron, run it everyday at 10AM local time: 0 10 * * *
// Author: Asep Norzai
// Twitter: @asepnorzai


import "@johnlindquist/kit"

let {data: {copyright, date, explanation, title, hdurl}} = await get(`https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY`)

show(`
	<div style="padding: 1em">
		<h1 style="text-align: center">Astronomy Picture of the Day</h1>
		<p><strong>Title:</strong> <a href="https://apod.nasa.gov/apod/astropix.html">${title}</a></p>
		<p><strong>Explanation:</strong> ${explanation}</p>
		<p><strong>Date:</strong> ${date}</p>
		<p><strong>Copyright:</strong> ${copyright}</p>
		<image src="${hdurl}" style="margin-top: 0.5em" title="${title}"/>
	</div>`,
{
	title,
	width: 800,
	height: 1000
})
```

## Extra Findings:

I initially wanted to use `// Schedule: 0 10 * * *` to automatically show the image everyday, 10am local time, but somehow when opening the script through `https://scriptkit.com/api/new?name=`, the `schedule` value gets stripped off, leaving only `// Schedule:`?
Not sure if it's a bug or by design?

**Comment by johnlindquist** [comment]
Oh, nice 😍  I'll have to try hooking this up to automatically set my desktop background.

## Re: Stripping Schedule Metadata

All metadata that can automatically run a script (Schedule/Background/System/Watch) gets stripped. This is by design to prevent a script from being run before the user has a chance to read through it.

Maybe renaming to something that doesn't run like `// Suggested Schedule: 0 10 * * *` would make more sense than stripping it completely 🤔

**Reply by Pashtunzoy** [reply]
I see why that makes sense now.
Also, added the suggested schedule, thanks.
